src/protocol/control_layer/ControlMessageV0Factory.js

```diff
@@ -0,0 +1,29 @@
+import UnsupportedVersionError from '../../errors/UnsupportedVersionError'
+import PublishRequestV0 from './publish_request/PublishRequestV0'
+import SubscribeRequestV0 from './subscribe_request/SubscribeRequestV0'
+import UnsubscribeRequestV0 from './unsubscribe_request/UnsubscribeRequestV0'
+import ResendRequest from './resend_request/ResendRequestV0'
+
+const messageClassByMessageType = {
+    publish: PublishRequestV0,
+    subscribe: SubscribeRequestV0,
+    unsubscribe: UnsubscribeRequestV0,
+    resend: ResendRequest,
+}
+
+export default class ControlMessageV0Factory {
```

Comment:
This separate class not needed anymore, if `ControlMessageFactory` assumes version is `0` for non-arrays?